centaur/src/it/scala/centaur/AbstractCentaurTestCaseSpec.scala

```diff
@@ -49,10 +47,21 @@ abstract class AbstractCentaurTestCaseSpec(cromwellBackends: List[String]) exten
 
   private def upgradeTestWdl(testCase: CentaurTestCase): CentaurTestCase = {
     import womtool.WomtoolMain
+    import better.files.File
 
     // The suffix matters because WomGraphMaker.getBundle() uses it to choose the language factory
-    val tempFile: path.Path = DefaultPathBuilder.createTempFile(suffix = "wdl").append(testCase.workflow.data.workflowContent)
-    val upgradeResult = WomtoolMain.upgrade(tempFile.pathAsString)
+    val rootWorkflowFile = File.newTemporaryFile(suffix = "wdl").append(testCase.workflow.data.workflowContent)
+    val workingDir: File = File.newTemporaryDirectory(prefix = testCase.workflow.testName)
+
+    rootWorkflowFile.copyTo(workingDir)
+
+    testCase.workflow.data.zippedImports match {
+      case Some(importsZip: File) =>
+        importsZip.unzipTo(workingDir)
+      case None => ()
+    }
+
+    val upgradeResult = WomtoolMain.upgrade(rootWorkflowFile.pathAsString)
```

Comment:
Do you not need to recurse into the imported files and import those too?

Reply:
I'm not sure if any of my test cases have transitive imports; I suppose that will become self-evident in Travis

Reply:
but even if they're not transitive, if x (draft-2) imports y (draft-2) and x gets upgraded to 1.0, does it not have problems importing y?

Reply:
This is addressed in the latest commits and the upgrade test in Travis passes